dhis-2/dhis-support/dhis-support-expression-parser/src/main/java/org/hisp/dhis/parser/expression/literal/SqlLiteral.java

```diff
@@ -59,4 +59,9 @@ public Object getBooleanLiteral( BooleanLiteralContext ctx )
     {
         return ctx.getText();
     }
+
+    private String escapeSql( String str )
+    {
+        return str == null ? null : StringUtils.replace( str, "'", "''" );
```

Comment:
Not sure exactly where and for what this code is used but is there potential for SQL injection with this approach?

Reply:
Not sure but we do have `org.hisp.dhis.system.util.SqlUtils#escapeSql`

Reply:
This code is used to create sql from expressions.
In this case, this escape function I think is needed to have the correct SQL more than SQL injection. But I am not sure 100%